domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/timeconversionutils/TimeConversionUtils.java

```diff
@@ -0,0 +1,99 @@
+ *  limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.domain.entity.timeconversionutils;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Utility class to carry out operations related to GTFS TYPE type
+ */
+public class TimeConversionUtils {
+    final static String pattern = "([0-9][0-9]|[0-9]):[0-5][0-9]:[0-5][0-9]";
+
+    /**
+     * Class constructor. Constructor is private to avoid instantiation of class.
+     */
+    private TimeConversionUtils() {
+    }
+
+    /**
+     * This method converts a time formatted as HH:MM:SS to a number of seconds elapsed since noon.
+     * The time is measured from "noon minus 12h" of the service day (effectively midnight except for days on which
+     * daylight savings time changes occur).
+     * @param timeAsString the time formatted as HH:MM:SS to convert as  a number of seconds elapsed since noon of day
+     *                     of service.
+     * @return  the number of seconds elapsed since noon of day of service. The time is measured from "noon minus 12h"
+     * of the service day (effectively midnight except for days on which daylight savings time changes occur).
+     */
+    public static Integer convertHHMMSSToIntFromNoonOfDayOfService(final String timeAsString) {
+        if (timeAsString != null && timeAsString.matches(pattern)) {
+            final String[] timeStringSplit = timeAsString.split(":");
+            try {
+                final int hourValue = Integer.parseInt(timeStringSplit[0]);
+                final int minuteValue = Integer.parseInt(timeStringSplit[1]);
+                final int secondValue = Integer.parseInt(timeStringSplit[2]);
+                // Converting time to an Integer value representing number of seconds
+                final int timeValueAsInt = hourValue * 3600 + minuteValue * 60 + secondValue;
+                // Setting noon as point zero, subtracting 12 * 3600 seconds
+                return timeValueAsInt - (12 * 3600);
+            } catch (ArrayIndexOutOfBoundsException | NumberFormatException e) {
+                return null;
+            }
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * This method converts a number of seconds elapsed since noon of say of service to a string formatted as HH:MM:SS.
+     *
+     * @param timeAsInt  the number of seconds elapsed since noon of say of service to be converted as a string
+     *                   formatted as HH:MM:SS.
+     * @return  the human readable string representation of the number of seconds elapsed since noon of day of service.
+     * This string is formatted as follows: HH:MM:SS.
+     */
+    public static String convertIntegerToHMMSS(final Integer timeAsInt) {
+        if (timeAsInt != null) {
+            int hourValue = (int) (TimeUnit.SECONDS.toHours(timeAsInt) + 12);
+            int minuteValue;
+            int secondValue;
+            if (timeAsInt > 0) {
```

Comment:
This could be a god solution to reduce the amount of code to read. For now this implementation does not seem to work, however I'll take a close look. 

Leaving this discussion open until we find a consensus :)